game_engine.py

```diff
@@ -0,0 +1,242 @@
+import json
+import sys
+import shutil
+import os
+import atexit
+
+
+class Game:
+    def __init__(self, map_file, game_window):
+        self.map_file = map_file
+        self.data = self.load_map(map_file)
+        self.game_window = game_window
+
+        if not self.data:
+            sys.exit('Error loading map file.')
+        atexit.register(self.quit_game)
+
+        self.player = Player(self.data, self)
+        self.current_room = Room(
+            self.data["rooms"].get(self.data["currentRoom"]), self.data, self)
+        
+        self.print(self.data['openning'])
+        self.current_room.describe_room()
+
+        self.command_palette = {
+            'north': self.move_player,
+            'south': self.move_player,
+            'west': self.move_player,
+            'east': self.move_player,
+            'take': self.player.pick_up_item,
+            'use': self.player.use_item,
+            'look': self.look_around,
+            'inventory': self.player.check_inventory,
+            'help': self.print_help,
+            'restart': self.restart_game,
+            'quit': self.quit_game,
+            'menu': self.go_to_menu
+        }
+
+    """If there isn't a copy already present,
+    create one. Read copied file and load it's contents
+    game_data"""
+    def load_map(self, filename):
+        self.copy_filename = filename.replace('maps/', 'tmp_')
+        if os.path.exists(self.copy_filename):
+            try:
+                with open(self.copy_filename, 'r') as f:
+                    game_data = json.load(f)
+                    return game_data
+            except json.JSONDecodeError:
+                sys.exit(f'Issue with copy_{filename}')
+        else:
+            try:
+                shutil.copyfile(filename, self.copy_filename)
```

Comment:
Sugar activities must be careful to write only to the activity root.  When installed in `/usr/share/sugar/activities `the activity won't be able to write into `maps/`.

Why is a file being copied?

Reply:
I overlooked that and will address that. The .json file is being copied and a .tmp file is created to be used for data storing. Said file, is updated to reflect the game's state